quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java

```diff
@@ -0,0 +1,116 @@
+package com.upgrad.quora.api.controller;
+
+import com.upgrad.quora.api.model.SigninResponse;
+import com.upgrad.quora.api.model.SignoutResponse;
+import com.upgrad.quora.api.model.SignupUserRequest;
+import com.upgrad.quora.api.model.SignupUserResponse;
+import com.upgrad.quora.service.business.UserAuthenticationService;
+import com.upgrad.quora.service.entity.UserAuthEntity;
+import com.upgrad.quora.service.entity.UserEntity;
+import com.upgrad.quora.service.exception.AuthenticationFailedException;
+import com.upgrad.quora.service.exception.SignOutRestrictedException;
+import com.upgrad.quora.service.exception.SignUpRestrictedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Base64;
+
+@RestController
+@RequestMapping("/")
+public class UserController {
+
+    @Autowired private UserAuthenticationService userAuthService;
+
+    /**
+     * Method for signing up a new user.
+     * An object of 'SignupUserRequest' is received as an argument with corresponding fields.
+     *
+     * @return SignupUserResponse - UUID of the new user created.
+     * @throws SignUpRestrictedException - Thrown if the username of email already exists in the DB.
+     */
+    @RequestMapping(
+            method = RequestMethod.POST,
+            path = "/user/signup",
+            consumes = MediaType.APPLICATION_JSON_UTF8_VALUE,
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<SignupUserResponse> signup(SignupUserRequest signupUserRequest)
+            throws SignUpRestrictedException {
+
+        UserEntity userEntity = new UserEntity();
+        userEntity.setFirstName(signupUserRequest.getFirstName());
+        userEntity.setLastName(signupUserRequest.getLastName());
+        userEntity.setUserName(signupUserRequest.getUserName());
+        userEntity.setEmail(signupUserRequest.getEmailAddress());
+        userEntity.setPassword(signupUserRequest.getPassword());
+        userEntity.setCountry(signupUserRequest.getCountry());
+        userEntity.setAboutMe(signupUserRequest.getAboutMe());
+        userEntity.setDob(signupUserRequest.getDob());
+        userEntity.setRole("nonadmin");
+        userEntity.setContactNumber(signupUserRequest.getContactNumber());
```

Comment:
I think we can create a private utility method which will take signupUserRequest and return us a userEntity, I guess that would be more cleaner

Reply:
Done. Added a separate method.